HerPortal.ManagementShell/Program.cs

```diff
@@ -0,0 +1,228 @@
+                        }
+                    }
+                }
+                catch (Exception e)
+                {
+                    outputProvider.Output($"Rollback following error in transaction: {e.InnerException?.Message}");
+                    dbContextTransaction.Rollback();
+                }
+            }
+
+            public void CreateUserOrLogError(string userEmailAddress, string[]? custodianCodes)
+            {
+                using var dbContextTransaction = dbContext.Database.BeginTransaction();
+                try
+                {
+                    var newLaUser = new User
+                    {
+                        EmailAddress = userEmailAddress,
+                        HasLoggedIn = false,
+                        LocalAuthorities = custodianCodes!
+                            .Select(code => dbContext.LocalAuthorities
+                                .Single(la => la.CustodianCode == code))
+                            .ToList()
+                    };
+                    dbContext.Add(newLaUser);
+                    dbContext.SaveChanges();
+                    outputProvider.Output("Operation successful");
+                    dbContextTransaction.Commit();
+                }
+                catch (Exception e)
+                {
+                    outputProvider.Output($"Rollback following error in transaction: {e.InnerException?.Message}");
+                    dbContextTransaction.Rollback();
+                }
+            }
+
+            public void RemoveLasFromUser(string[]? custodianCodes, User? user)
+            {
+                var lasToRemove = user?.LocalAuthorities.Where(la => custodianCodes!.Contains(la.CustodianCode)).ToList();
+                using var dbContextTransaction = dbContext.Database.BeginTransaction();
+                try
+                {
+                    if (lasToRemove != null)
+                        foreach (var la in lasToRemove)
+                        {
+                            user?.LocalAuthorities.Remove(la);
+                        }
+
+                    dbContext.SaveChanges();
+                    outputProvider.Output("Operation successful");
+                    dbContextTransaction.Commit();
+                }
+                catch (Exception e)
+                {
+                    outputProvider.Output($"Rollback following error in transaction: {e.InnerException?.Message}");
+                    dbContextTransaction.Rollback();
+                }
+            }
+
+            public void AddLasToUser(string[]? custodianCodes, User? user)
+            {
+                if (custodianCodes == null) return;
+                using var dbContextTransaction = dbContext.Database.BeginTransaction();
+                try
+                {
+                    foreach (var code in custodianCodes)
+                    {
+                        var la = dbContext.LocalAuthorities.SingleOrDefault(la => la.CustodianCode == code);
```

Comment:
We don't make any kind of check that the custodian code has correctly found an LA - can we add something like that?

Reply:
Wrapped in a try/catch